be1-go/channel/election/mod.go

```diff
@@ -212,47 +210,57 @@ func (c *Channel) Catchup(catchup method.Catchup) []message.Message {
 }
 
 // Broadcast is used to handle a broadcast message.
-func (c *Channel) Broadcast(broadcast method.Broadcast, socket socket.Socket) error {
+func (c *Channel) Broadcast(broadcast method.Broadcast, _ socket.Socket) error {
 	err := c.verifyMessage(broadcast.Params.Message)
 	if err != nil {
 		return xerrors.Errorf("failed to verify broadcast message on an "+
 			"election channel: %w", err)
 	}
 
-	err = c.handleMessage(broadcast.Params.Message, socket)
+	msg := broadcast.Params.Message
+	data := msg.Data
+
+	jsonData, err := base64.URLEncoding.DecodeString(data)
+	if err != nil {
+		return xerrors.Errorf("failed to decode message data: %v", err)
+	}
+
+	object, _, err := messagedata.GetObjectAndAction(jsonData)
 	if err != nil {
-		return xerrors.Errorf("failed to handle broadcast message: %v", err)
+		return xerrors.Errorf("failed to get object and action from message data: %v", err)
+	}
+
+	if object == messagedata.ElectionObject {
```

Comment:
I do not get why it is necessary to do this check now and not in master

Reply:
See previous comment. The tests expected receiving a non election message shouldn't return an error.